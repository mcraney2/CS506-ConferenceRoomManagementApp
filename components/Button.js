// ./components/Button.js
import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";

const Button = props => {
  return (
        <TouchableOpacity onPress={props.handleClick}>
            <View style={styles.button}>
                <Text style={styles.buttonText}>
                    {props.label}
                </Text>
            </View>
        </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
    button: {
        backgroundColor: 'red',
        paddingVertical: 12,
        paddingHorizontal: 25,
        borderRadius: 25,
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonText: {
        color: "white",
        fontSize: 18
    }
});

export default Button;

/*
<button className={className} onClick={props.handleClick}>
        
        {props.label}
    </button>
    */