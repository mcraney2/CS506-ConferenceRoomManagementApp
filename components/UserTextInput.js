import React, { Component } from 'react';
import { TextInput } from 'react-native-gesture-handler';

class UserTextInput extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            height: 30,
            
            width: 250,
            borderColor: "grey",
            borderWidth: 1,
            margin: 10,
        }
    }
    updateSize = (height) => {
        this.setState({
          height
        });
      }
    render() { 
        const newValue = this.props.value;
        const height = this.state;
        let newStyle = height;
        return (  
        <TextInput
            placeholder= {this.props.placeHolder}
            onChangeText={(value) => this.props.setValue(value)}
            style={newStyle}
            editable={true}
            multiline={true}
            value={newValue}
            testID = 'text'
            onContentSizeChange={(e) => this.updateSize(e.nativeEvent.contentSize.height)}
          />);
    }
}
 
export default UserTextInput;