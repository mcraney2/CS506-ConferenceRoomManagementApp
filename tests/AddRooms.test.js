import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddRoom from '../components/AddRooms';
import { TextInput } from 'react-native-gesture-handler';
import UserTextInput from '../components/UserTextInput';
import Button from '../components/Button';


test('Renders AddRoom snapshot as expected', () => {
    const tree = renderer.create(<AddRoom/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Room test input sends room number to handler', () => {
    const onEventMock = jest.fn();
    const {getByPlaceholderText} = render(
      <UserTextInput setValue = {onEventMock} placeHolder = 'Enter Room Number'/>
    );
    fireEvent(getByPlaceholderText('Enter Room Number'), 'onChangeText','123');
    expect(onEventMock).toHaveBeenCalledWith('123');
});

test('Button sets state of room number', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<AddRoom/>);
    const instance = test.instance();
    instance.setRoomNum('1234')
    expect(test.state('roomNum')).toBe('1234');
});

