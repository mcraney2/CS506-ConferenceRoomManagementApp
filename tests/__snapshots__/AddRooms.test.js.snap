// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders AddRoom snapshot as expected 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    editable={true}
    multiline={true}
    onChangeText={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholder="Conference Room Name/Number"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "grey",
        "borderWidth": 1,
        "height": 30,
        "margin": 10,
        "width": 250,
      }
    }
    testID="text"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 25,
          "justifyContent": "center",
          "paddingHorizontal": 25,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Add
      </Text>
    </View>
  </View>
</View>
`;
